Loader/ComboLoader.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * File containing the ComboLoader class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformUIBundle\Loader;
+
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+
+/**
+ * ComboLoader combines JS or CSS files.
+ */
+class ComboLoader implements Loader
+{
+    /**
+     * @var string Path to YUI build directory.
+     */
+    private $yuiBuildDirectory;
+
+    /**
+     * @var string Path to web directory.
+     */
+    private $webDirectory;
+
+    /**
+     * @param string $yuiBuildDirectory
+     * @param string $webDirectory
+     */
+    public function __construct($yuiBuildDirectory, $webDirectory)
+    {
+        $this->yuiBuildDirectory = $yuiBuildDirectory;
+        $this->webDirectory = $webDirectory;
+    }
+
+    /**
+     * Returns combined JS or CSS files.
+     *
+     * @param array $files File name array
+     *
+     * @return string Combined files
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\Core\Base\Exceptions\NotFoundException
+     */
+    public function getContent(array $files)
+    {
+        $this->assertAtLeastOneFile($files);
+
+        $content = '';
+
+        foreach ($files as $file => $value) {
+            $ext = $this->findFileExtension($file);
+
+            if ($ext === 'js') {
+                if ($this->isYuiFile($file)) {
+                    $file = $this->fixUnresolvedPath($file);
+                }
+
+                $content .= $this->loadFile($file, $ext);
+            } elseif ($ext === 'css') {
+                $file = $this->fixUnresolvedPath($file);
+
+                $content .= $this->loadFile($file, $ext);
+            } else {
+                throw new NotFoundException('file type', $file);
+            }
+        }
+
+        return $content;
+    }
+
+    /**
+     * Returns content type.
+     *
+     * @param array $files File name array
+     *
+     * @return string
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\Core\Base\Exceptions\NotFoundException
+     */
+    public function getContentType(array $files)
+    {
+        $this->assertAtLeastOneFile($files);
+
+        $file = key($files);
```

Comment:
Well, this is because `files` array has file names stored in keys. But I could be a nice idea to swap keys and values in the controller and pass them to loader in values.